new-log-viewer/src/services/LogFileManager/index.ts

```diff
@@ -254,6 +277,80 @@ class LogFileManager {
         };
     }
 
+    /**
+     * Searches for log events based on the given search string.
+     *
+     * @param searchString
+     * @param isRegex
+     * @param isCaseSensitive
+     */
+    startQuery (searchString: string, isRegex: boolean, isCaseSensitive: boolean): void {
+        // If the search string is empty, or there are no logs, return
+        if ("" === searchString || 0 === this.#numEvents) {
+            return;
+        }
+
+        // Construct search RegExp
+        const regexPattern = isRegex ?
+            searchString :
+            searchString.replace(/[.*+?^${}()|[\]\\]/g, "\\$&");
+        const regexFlags = isCaseSensitive ?
+            "" :
+            "i";
+        const searchRegex = new RegExp(regexPattern, regexFlags);
+
+        this.#searchChunkAndScheduleNext(this.#queryId++, 0, searchRegex);
```

Comment:
As discussed offline, we don't want to increment the ID after the search task is scheduled, and it makes sense to increment the ID for every new query including the empty queries.
```suggestion
        this.#queryId++;

        // If the search string is empty, or there are no logs, return
        if ("" === searchString || 0 === this.#numEvents) {
            return;
        }

        // Construct search RegExp
        const regexPattern = isRegex ?
            searchString :
            searchString.replace(/[.*+?^${}()|[\]\\]/g, "\\$&");
        const regexFlags = isCaseSensitive ?
            "" :
            "i";
        const searchRegex = new RegExp(regexPattern, regexFlags);

        this.#searchChunkAndScheduleNext(this.#queryId, 0, searchRegex);
```